flink-clients/src/test/java/org/apache/flink/client/program/PerJobMiniClusterFactoryTest.java

```diff
@@ -190,6 +190,7 @@ private static JobGraph getNoopJobGraph() {
     private static JobGraph getCancellableJobGraph() {
         JobVertex jobVertex = new JobVertex("jobVertex");
         jobVertex.setInvokableClass(MyCancellableInvokable.class);
+        jobVertex.setParallelism(1);
```

Comment:
why not just do this in the JobGraphBuilder?

Reply:
I thought about that decided against it: I first assumed `JobGraphBuilder` being a testing class. But realizing that it's part of the production code I hesitated to change the class' contract to fix test issues.

Reply:
The downside though is that you need to touch _every single relevant test_, whereas the `JobGraphBuilder` approach would cover most(all?) of them.
FLINK-21866 shows yet another test that fails for the same reason.

Reply:
You're right. There are numerous tests that are affected by this. I took another look into `JobGraphBuilder`. @rmetzger is it save to assume that a `Streaming` job is not allowed to have the parallelism set to `-1`?

In general, there seem to be some ongoing refactorings with the `JobGraphBuilder`? The only time it is used in production is for [creating a Batch JobGraph](https://github.com/apache/flink/blob/e000723fba5ff76d5c837e49d16c570dc55c274a/flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraphBuilder.java#L142). The streaming jobs seem to rely on `StreamGraphGenerator` which does not utilize the `JobGraphBuilder`.

It feels like `StreamGraphGenerator` would be the most appropriate location to set the parallelism if we say that this is a general requirement for Streaming `JobGraph`s. But that would leave us with the problem that it's not implemented in the `JobBuilder` and requires us to adapt quite a few tests accordingly.

Reply:
I'm always careful changing default parallelisms in production code, as there could be unforeseen effects (incompatible savepoints across versions, parallelism changes after upgrading to Flink 1.13 in scenarios where users rely on this somehow).

From this PR it seems that the number of tests that violate the contract seems reasonably small to just fix the tests.

Reply:
Ok. I enabled adaptive scheduling now (I didn't realize that it wasn't done in the first place). And fix the failing tests one by one.

Reply:
It's only enabled for the nightly tests. But I guess by now all issues by my change have been reported 😢 . I just wanted to do a final verification run to be sure.

Thanks a lot!